tests/core/actionrun_test.py

```diff
@@ -687,6 +687,10 @@ def test_is_run_blocked_required_actions_failed(self):
         self.run_map['action_name'].machine.state = ActionRun.STATE_FAILED
         assert self.collection._is_run_blocked(self.run_map['second_name'])
 
+    def test_is_run_blocked_required_actions_missing(self):
+        del self.run_map['action_name']
```

Comment:
does this mutate self.run_map for the rest of the test class? might be worth cloning the run_map before mutating,

Reply:
No, that's the magic of testify! :) The @setup method that creates the run_map runs before every test case.